doc/otn/otn_otss_hld.md

```diff
@@ -0,0 +1,208 @@
+# Optical Transport State Service (OTSS) High Level Design #
+
+## Table of Content 
+
+### Revision  
+Rev | Rev	Date	| Author	| Change Description
+---------|--------------|-----------|-------------------
+|v0.1 |05/03/2024  |Weitang Zheng | Initial version
+
+### Scope  
+
+This document describes the high level design of OTSS in SONiC
+
+### Definitions/Abbreviations 
+
+Definitions/Abbreviation|Description
+------------------------|-----------
+OTSS| Optical Transport State Service
+OTAI| Optcial Transport Abstraction Interface
+SWSS| Switch State Service
+OTN| Optical Transport Network
+CRUD| CREATE, READ, UPDATE and DELETE
+OA| Optical Amplifier
+OSC| Optical Supervisory Channel
+OLP| Optical Line Protection
+VOA|Optical Attenuator
+OTDR| Optical Time Domain Reflectometer
+FSM| Finite-State Machine
+UUID| Universally Unique IDentifier
+
+
+
+### Overview 
+
+OTSS is a SONiC user space docker application, which manages all the optical components and object in an OTN device with an unified and vendor-neutral OTAI APIs. It's in parallel with the SWSS and SAI in the SONiC switch.
```

Comment:
Creating a parallel infrastructure for SWSS technically defeat some of the core objectives of Sonic. The code sharing and all the benefits of unified testing.